massa-grpc/src/stream/new_operations.rs

```diff
@@ -0,0 +1,177 @@
+use crate::error::GrpcError;
+use crate::service::MassaGrpcService;
+use futures_util::StreamExt;
+use massa_models::operation::{OperationType, SecureShareOperation};
+use massa_proto::massa::api::v1::{
+    self as grpc, BytesMapFieldEntry, NewOperationsStreamRequest, NewOperationsStreamResponse,
+    OperationStreamFilterType,
+};
+use std::pin::Pin;
+use tokio::select;
+use tonic::codegen::futures_core;
+use tonic::{Request, Streaming};
+use tracing::log::error;
+
+/// Type declaration for StreamTransactionsThroughputStream
+pub type NewOperationsStream = Pin<
+    Box<
+        dyn futures_core::Stream<Item = Result<NewOperationsStreamResponse, tonic::Status>>
+            + Send
+            + 'static,
+    >,
+>;
+
+/// Creates a new stream of new produced and received operations
+pub(crate) async fn new_operations(
+    grpc: &MassaGrpcService,
+    request: Request<Streaming<NewOperationsStreamRequest>>,
+) -> Result<NewOperationsStream, GrpcError> {
+    // Create a channel to handle communication with the client
+    let (tx, rx) = tokio::sync::mpsc::channel(grpc.grpc_config.max_channel_size);
+    // Get the inner stream from the request
+    let mut in_stream = request.into_inner();
+    // Subscribe to the new operations channel
+    let mut subscriber = grpc.pool_channels.operation_sender.subscribe();
+
+    // Spawn a new task for sending new operations
+    tokio::spawn(async move {
+        if let Some(Ok(request)) = in_stream.next().await {
```

Comment:
IMHO, It's always empty

Reply:
here, it's waiting for the filter

Reply:
the task is here to send and update filter, the sub comments are clearer